posthog/hogql_queries/insights/trends/aggregation_operations.py

```diff
@@ -0,0 +1,147 @@
+from typing import List
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_expr, parse_select
+from posthog.hogql_queries.utils.query_date_range import QueryDateRange
+from posthog.schema import ActionsNode, EventsNode
+
+
+class QueryAlternator:
+    _query: ast.SelectQuery
+    _selects: List[ast.Expr]
+    _group_bys: List[ast.Expr]
+    _select_from: ast.JoinExpr | None
+
+    def __init__(self, query: ast.SelectQuery):
+        self._query = query
+        self._selects = []
+        self._group_bys = []
+        self._select_from = None
+
+    def build(self) -> ast.SelectQuery:
+        if len(self._selects) > 0:
+            self._query.select.extend(self._selects)
+
+        if len(self._group_bys) > 0:
+            if self._query.group_by is None:
+                self._query.group_by = self._group_bys
+            else:
+                self._query.group_by.extend(self._group_bys)
+
+        if self._select_from is not None:
+            self._query.select_from = self._select_from
+
+        return self._query
+
+    def appendSelect(self, expr: ast.Expr) -> None:
+        self._selects.append(expr)
+
+    def appendGroupBy(self, expr: ast.Expr) -> None:
+        self._group_bys.append(expr)
+
+    def replaceSelectFrom(self, join_expr: ast.JoinExpr) -> None:
+        self._select_from = join_expr
+
```

Comment:
Our adherence to this is a mixed bag... but normally python prefers `snake_case` method names

Reply:
Ahh, that's me falling into JS habits 🤦 Not intentional - will change